game/src/OptionMenu/ResolutionSelector.gd

```diff
@@ -63,7 +71,47 @@ func _set_value_from_file(load_value):
 
 func _on_item_selected(index : int):
 	if _valid_index(index):
+		previous_resolution = Resolution.get_current_resolution()
 		Resolution.set_resolution(get_item_metadata(index))
+		var new_resolution = get_item_metadata(index)
+		#emit a signal if this option was changed by the user
+		if was_pressed and index != previous_selected_index:
+			emit_signal("selection_changed",previous_resolution,new_resolution)
 	else:
 		push_error("Invalid ResolutionSelector index: %d" % index)
 		reset_setting()
+
+#Dialog revert functionality below here
+#UIFUN-301 and UIFUN-302
+
+#This is just to filter out cases where _on_item_selected is not called by the
+#user changing a setting (which happens on startup)
+var was_pressed:bool = false
+func _on_toggled(button_pressed):
```

Comment:
`button_pressed` is already a variable in BaseButton, please avoid shadowing variables as it creates warnings in the editor. (and can create problems down the line, hence why its a warning)